Adding extra SEPA fields: Purpose & Schema

andreparames opened the pull request:
This commit adds two standard SEPA fields to the generator, the Purpose Code and the Creditor Schema:

https://wiki.xmldation.com/Support/Nordea/Purpose_Code

https://wiki.xmldation.com/Support/EPC/CdtrSchmeId

alexis-via committed:
Store field acc_type on res.partner.bank, so that we can search and groupby on it

alexis-via committed:
Better handling of partner_bank_id and mandate_id on invoice: invisible/required/onchange

alexis-via committed:
Fix crash in mandate report

sbidoul committed:
Avoid confusion between payment mode and payment method

alexis-via committed:
Move field bank_account_required from module account_payment_partner to account_payment_mode

Make the mandate a required field on payment line when the payment method has mandate_required=True
Make the bank account a required field on payment line when the payment method has bank_account_required=True
Minor code cleanup
PEP8

alexis-via committed:
FIX manual payment method

alexis-via committed:
Enable the payment methods by default on bank journals (including existing bank journals via post_install scripts)

Enable "Generate Accounting Entries On File Upload" by default

alexis-via committed:
Add a unit test

alexis-via committed:
Better test that include several invoices (to test grouping) and refunds

alexis-via committed:
Add unittest for SEPA direct debit

Commit:
Merge pull request #3 from sbidoul/patch-1

Avoid confusion between payment mode and payment method

sbidoul committed:
[IMP] prevent deleting a move-line that is linked to a payment order line

sbidoul committed:
[FIX] better filtering of move lines already included in open payment order

Commit:
Merge pull request #5 from acsone/9.0-fix-line-exclusion-sbi

[FIX] better filtering of move lines already included in open payment order

Commit:
Merge pull request #4 from acsone/9.0-restrict-move-line-delete-sbi

[IMP] prevent deleting a move-line that is linked to a payment order line

sbidoul committed:
Merge pull request #270 from akretion/9-sorrento-code-sprint

[9.0][2nd revolution] port to v9 with several improvements (sorrento code sprint)

OCA-git-bot committed:
[UPD] addons table in README.md

alexis-via committed:
Add support for multi-currency in the generation of the account move.

Take into account all the remaining remarks of Frederic Clementi dated May 30:
- see multi-currency amounts in the wizard to add transaction lines
- add help msg on bank payment lines

alexis-via committed:
Allow to link a bank journal to an existing bank account (one of the remarks of Frederic Clementi)

alexis-via committed:
Add unitest for a USD credit transfer (in EUR company)

fclementic2c committed:
Merge pull request #278 from akretion/9-several-improvements

9.0 Add support for multi-currency in the generation of the account move

pedrobaeza committed:
[IMP] account_payment_order: Remove useless migration script

pedrobaeza committed:
[IMP] Remove unused POT files

pedrobaeza committed:
[IMP] account_banking_sepa_direct_debit: Spanish translation

pedrobaeza committed:
[FIX] account_banking_sepa_direct_debit: Fix mandate report layout

oca-transbot committed:
OCA Transbot updated translations from Transifex

yvaucher committed:
Move the account_payment_mode ir.rule in account_payment_mode module

antespi committed:
[IMP] Add english lang when a different lang is printed

antespi committed:
English text at right instead of below

antespi committed:
Fit in order to have only one page per mandate

antespi committed:
Fix t-value when language is False

pedrobaeza committed:
Merge pull request #280 from Tecnativa/9.0-imp_direct_debit_mandate_native_lang

[IMP] Add english lang when a different lang is printed

sbidoul committed:
[IMP] account_payment_order: extensibility of _prepare_move

Pass the list of bank lines to _prepare_move so it is possible to customize the move (eg it's name) based on the lines being paid.

pedrobaeza committed:
Merge pull request #285 from OCA/sbidoul-patch-1

[IMP] account_payment_order: extensibility of _prepare_move

sbidoul committed:
[FIX] payment order: add missing onchange dependencies in move selection wizard

sbidoul committed:
[FIX] payment order: remove duplicate in previous commit

pedrobaeza committed:
Merge pull request #287 from acsone/9.0-onchange-depends-sbi

[FIX] payment order: add missing onchange dependencies in move selection wizard

alexis-via committed:
FIX crash when selecting a customer that has a direct debit payment mode but no mandate

Hide many fields when payment_order_ok is False on a payment_mode
Fix payment line communication when the move line is not linked to an invoice (problem found when porting donation_debit_debit to v9)

alexis-via committed:
For the communication field, if self.move_id.ref is False, fallback on self.move_id.name

pedrobaeza committed:
Merge pull request #290 from akretion/9small_usability_improvements

9.0 FIX crash when selecting a customer that has no mandate + small improvements

oca-transbot committed:
OCA Transbot updated translations from Transifex

jesusVMayor committed:
account_payment_purchase: set correctly payment mode in purchases created from procurement.

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[IMP] set partner on offsetting account move line if possible

fixes #277

pedrobaeza committed:
[FIX] .travis.yml: Correct indentation

pedrobaeza committed:
Merge pull request #294 from acsone/9.0-offsetting-partner-sbi

[IMP] set partner on offsetting account move line if possible

sbidoul committed:
bump minor version of account_payment_order

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
Merge pull request #279 from yvaucher/9.0-account_payment_mode-ir_rule

[9.0] Move the account_payment_mode ir.rule in account_payment_mode module

yvaucher committed:
Reactivate company rules in account_payment_order

sbidoul committed:
Merge pull request #301 from yvaucher/uncomment-ir-rules

Reactivate company rules in account_payment_order

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

pedrobaeza committed:
[MIG] Remove setup module directories

pedrobaeza committed:
[MIG] Update metafiles

Commit:
Port almost all modules to v10 (#305)

Port almost all modules to v10

* Update to EPC Rulebook v9.2 that start to apply on 2016-11-20 (bug #300)

OCA-git-bot committed:
[ADD] setup.py

alexis-via committed:
Remove method which is not called anymore

OCA-git-bot committed:
[UPD] addons table in README.md

hbrunn committed:
Merge pull request #307 from akretion/10-code-cleanup

10.0 Remove method which is not called anymore

Commit:
[FIX] account_payment_order: Use 'index' instead of 'select'

Commit:
[FIX] account_payment_order: Use 'index' instead of 'select'

sbidoul committed:
Merge pull request #315 from adrienpeiffer/patch-2

[FIX] account_payment_order: Use 'index' instead of 'select'

sbidoul committed:
Merge pull request #314 from adrienpeiffer/patch-1

[FIX] account_payment_order: Use 'index' instead of 'select'

oca-transbot committed:
OCA Transbot updated translations from Transifex

jalzaga committed:
Remove internal_type=other restriction for transfer account (#311)

pedrobaeza committed:
Remove author as the contrib is only an small fix

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Creation/Modification of bank info only for Account Payment group

alexis-via committed:
FIX missing include of account_payment_mode/security/account_payment_mode.xml

sbidoul committed:
Merge pull request #336 from akretion/10-fix-multicompany

[10.0] FIX missing include of account_payment_mode/security/account_payment_mode.xml

vrenaville committed:
[FIX] fill date instead off maturity date on account move line generated

yvaucher committed:
Merge pull request #335 from vrenaville/payment_order_maturity_date

[FIX] fill date instead off maturity date on account move line generated

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
Merge pull request #331 from akretion/10-payment-security

[10.0] Creation/Modification of bank info only for Account Payment group

mdietrichc2c committed:
Fix 2 issues on form view for payment order

Commit:
Merge pull request #347 from yvaucher/10.0-fix-346

[10.0] [FIX] 2 issues on form view for payment order

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] addons table in README.md

pedrobaeza committed:
[CHG] *: Change copyright and author after company merging

pedrobaeza committed:
[IMP] account_payment_partner: Fill payment mode in invoices if none is provided

Using same method as in upstream, payment mode is filled on invoice creation if
no payment method is provided. This way, we don't need to install
account_payment_sale if we don't want to handle several payment modes at sales
level. Even more, if we install the module later and we have already existing
sales orders without payment mode filled, those orders will be invoiced with
the customer payment mode.

pedrobaeza committed:
[IMP] account_banking_mandate: Fill mandate in invoices if none is provided

Using same method as in upstream, mandate is filled on invoice creation if no
one is provided. This way, we don't need to install account_banking_mandate_sale
if we don't want to handle several mandates at sales level.

alexis-via committed:
[FIX] Bug #353 crash when selecting a payment mode that has a variable link to a bank account

cubells committed:
[9.0][MIG] account_payment_purchase module

pedrobaeza committed:
[FIX] account_banking_mandate/account_payment_partner: Signature changed in convert_to_write

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Add 'post_move' option on account.payment.mode

alexis-via committed:
post_move default True, to keep the previous behavior

Bump up version number

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
Merge pull request #363 from akretion/10-no_auto_post

[10.0] Add 'post_move' option on account.payment.mode

Commit:
Merge pull request #354 from akretion/10-fix-bug353

[FIX] Bug #353 crash when selecting a payment mode that has a variable link to bank accounts

OCA-git-bot committed:
[UPD] addons table in README.md

alexis-via committed:
Add field default_date_prefered on payment mode

alexis-via committed:
Update version number

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
[IMP] Explicit error when bank account is missing on bank journal

alexis-via committed:
Add support for ISO20022 "Category Purpose"

Add local instrument 'INST' for SEPA Instant Credit Transfer

pedrobaeza committed:
Merge pull request #366 from akretion/10-category-purpose

[10.0] Add support for ISO20022 "Category Purpose"

alexis-via committed:
Update version number

Commit:
Merge pull request #365 from akretion/10-better-error

[10.0] Explicit error when bank account is missing on bank journal

alexis-via committed:
'date_prefered' is a required field on payment orders: so, if 'date_prefered' key is present in vals but has no value, we use the default value on the payment mode

alexis-via committed:
FIX demo data: signature date of mandate was too old and was automatically expired

Commit:
Merge pull request #364 from akretion/10-default_date_prefered

[10.0] Add field default_date_prefered on payment mode

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

GillesTephaneMeyomesse committed:
[ADD] description field for payment order

grindtildeath committed:
ADD relational inverse field for payment.mode on payment.method

oca-transbot committed:
OCA Transbot updated translations from Transifex

mdietrichc2c committed:
[IMP] account_payment_order: Don't allow deletion of an uploaded payment order + allow to cancel an uploaded payment order

grindtildeath committed:
[IMP] account_payment_order: Restriction + tests

* Restrict the deletion of bank payment line
* Add test_cancel_payment_order

carlosdauden committed:
[IMP] account_banking_mandate: Add valid_mandate field and fix onchange

(cherry picked from commit 25d2e4d)

pedrobaeza committed:
[FIX] account_payment_order: Adapt test to v10

carlosdauden committed:
[IMP] account_banking_mandate: Changes suggested

(cherry picked from commit 45c1cc2)

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

GillesTephaneMeyomesse committed:
[MIG] add fr translation

Commit:
[IMP] account_payment_order: select partner bank account automatically.

pedrobaeza committed:
Merge pull request #380 from acsone/10.0-payment-order-bank-account-ape

[10.0][IMP] account_payment_order: select partner bank account automatically.

OCA-git-bot committed:
[UPD] addons table in README.md

pedrobaeza committed:
[IMP] account_banking*: Show identifiers at payment mode level

Now the visibility is controlled through a security group.

Commit:
Merge pull request #379 from acsone/10.0-fr_translation_mgs

[MIG][10] add fr translation

oca-transbot committed:
OCA Transbot updated translations from Transifex

astirpe committed:
[FIX][10.0] account_banking_mandate: onchange return value missing (#382)

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
Merge pull request #381 from Tecnativa/10.0-identifiers_per_payment_mode

[10.0][IMP] account_banking*: Show identifiers at payment mode level

OCA-git-bot committed:
[UPD] addons table in README.md

grindtildeath committed:
[10.0]Fix payment method onchange with multicompany (#374)

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
[UPD] if a mandate is linked to the bank account used, put it automatically on the payment line'

Commit:
Merge pull request #389 from acsone/10.0-banking_mandate-cpi

[10.0][account_banking_mandate] if a mandate is linked to the bank account used, put it automatically on the payment line

Commit:
Merge pull request #372 from acsone/10.0-add_description_payment_mgs

[ADD] description field for payment order

Commit:
Merge pull request #375 from grindtildeath/10.0_add_payment_mode_inverse_field

[10.0][ADD] account_payment_mode : relational inverse field for payment.mode on payment.method

OCA-git-bot committed:
[UPD] addons table in README.md

GillesTephaneMeyomesse committed:
[ADD] report for payment order (#370)

carlosdauden committed:
[IMP] account_banking_mandate: Fill payment line with first valid mandate

carlosdauden committed:
[FIX] account_payment_order: Sequences restarted when update module

Commit:
Merge pull request #412 from Tecnativa/10.0-account_payment_order-noupdate

[FIX] account_payment_order: Sequences restarted when update module

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

JordiBForgeFlow committed:
Merge pull request #410 from Tecnativa/10.0-IMP-account_banking_mandate

[IMP] account_banking_mandate: Fill payment line with first valid mandate

OCA-git-bot committed:
[UPD] addons table in README.md

carlosdauden committed:
[FIX] account_banking_mandate: Restore param removed by cherry-pick (#415)

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0][IMP] account_payment_order: pass default partner_id when creating bank directly in payment line

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[FIX] account_banking_mandate: If Create and edit acc_number not set

TDu committed:
[ADD] Bank name and bic to report_invoice

pedrobaeza committed:
Merge pull request #422 from Tecnativa/10.0-FIX-account_banking_mandate-quick_create_acc_number

[FIX] account_banking_mandate: If Create and edit acc_number not set

OCA-git-bot committed:
[UPD] addons table in README.md

TDu committed:
fixup! [ADD] Bank name and bic to report_invoice

cubells committed:
[IMP] account_payment_sale: add payment mode to sale report

Commit:
Merge pull request #420 from versada/imp_link_partner_bank_id

[10.0][IMP] account_payment_order: pass default partner_id when creaing bank directly in payment line

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[CHG] account_payment_partner: Prevent to delete bank account linked on invoices.

pedrobaeza committed:
Merge pull request #426 from Tecnativa/10.0-add_payment_mode_to_sale_report

[IMP] account_payment_sale: add payment mode to sale report

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

fclementic2c committed:
Merge pull request #423 from TDu/add-bank-info-on-invoice

[ADD] Bank name and bic to report_invoice

OCA-git-bot committed:
[UPD] addons table in README.md

cmtCousteau committed:
CO-1314 - Undo payment has been move to bank-payment

ecino committed:
Code refactoring of module account_payment_line_cancel

aisopuro committed:
[FIX] Ensure that the test finds only one bank journal

Previously it was possible to find multiple bank journals,
which caused singleton errors when the test tried to set values
on them.

acsonefho committed:
[FIX] Fix issue when the bank_id of partner_bank_id is not set (field not mandatory) and create a traceback for the user

Use t-if

Increment the version number

pedrobaeza committed:
Merge pull request #449 from acsonefho/fho/fix_account_payment_partner_invoice_report

10.0 [FIX] account_payment_partner: Invoice report

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
Merge pull request #437 from avoinsystems/ensure-single-bank-in-test

[FIX] Ensure that the test finds only one bank journal

NL66278 committed:
[FIX] account_banking_pain_base. Only unidecode unicode objects.

NL66278 committed:
[IMP] account_banking_pain_base. Increment module version.

pedrobaeza committed:
Merge pull request #454 from NL66278/10.0-fix-unidecode-warning

[10.0][FIX] account_banking_pain_base. Only unidecode unicode objects.

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] account_banking_sepa_credit_transfer: Fix tests due to upstream change

Odoo has added a constraint for avoiding a company currency change if there
are move lines, making these tests to fail, as the currency is changed to EUR.

With this commit, we create a new company with EUR currency for avoiding the
problem. An alternative solution could be to delete all the existing move lines
on the company, but as that can be problematic if there are additional
constraints (extra modules, locking date, etc), I have chosen the other approach.

pedrobaeza committed:
[FIX] account_banking_sepa_direct_debit: Fix tests due to upstream change

Odoo has added a constraint for avoiding a company currency change if there
are move lines, making these tests to fail, as the currency is changed to EUR.

With this commit, we create a new company with EUR currency for avoiding the
problem.

This commit also changes account_banking_mandate for not duplicating mandate
number, as it was detected during the test creation.

Similar to 1f8e345

Commit:
[10.0] Usability of account_payment_order: access journal entries via top-right button (#457)

account_payment_order: access journal entries via top-right button instead of dedicated tab

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] account_banking_mandate_sale: Fix error when no valid mandate for the partner

Closes #459

OCA-git-bot committed:
[UPD] addons table in README.md

pedrobaeza committed:
[FIX] account_banking_sepa_direct_debit: Avoid problems in DBs with other localizations

If you have another localization installed, this test is not going to work, as the
defaults assign other taxes as defaults, and the generic CoA is marked as not complete
set, so values are not set either on onchange.

Commit:
[10.0][FIX] Split account move generated for SEPA direct debits per sequence_type (#464)

The sequence_type of mandates switches from First to Recurring:
- BEFORE: when state switched from 'open' to 'generated'
- NOW: when state switches from 'generated' to 'uploaded'

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
[FIX] account_payment_mode: Don't store acc_type of res.partner.bank (#467)

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py [ci skip]

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] account_payment_order: Take payment mode from move (#466)

When adding to a payment order from invoices, the payment mode
considered is the one in the invoice, but once you have confirmed it,
you can't change it, so if you change the payment mode in the journal
items, it doesn't make the difference.

With this change, this is taken into account for adding to the
payment order(s) properly.

OCA-git-bot committed:
[UPD] addons table in README.md [ci skip]

hbrunn committed:
[FIX] only drop column if it exists

pedrobaeza committed:
Merge pull request #474 from hbrunn/10.0-account_payment_mode-migration_drop_column_if_exists

[FIX] only drop column if it exists

carlosdauden committed:
[FIX] account_payment_order: Add bank payment line multi-company rule

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Several small usability improvements

- when manually encoding a payment line, get company currency as default currency
- Search on payment order numbers
- Direct search on bank journal name in payment orders
- remove widget="selection" on account.payement.mode : this object is rarely modified, so widget='selection' is not a time saver, but we miss the HTML link, which can be pretty convenient to check the configuration.

pedrobaeza committed:
Merge pull request #476 from Tecnativa/10.0-FIX-account_payment_order-security_rule

[FIX] account_payment_order: Add bank payment line multi-company rule

pedrobaeza committed:
Merge pull request #479 from akretion/10-small-usability-improvements

[10.0] Several small usability improvements

lembregtse committed:
[FIX] account_banking_mandate: Add signature_date for _check_valid_state constraint (#445)

pedrobaeza committed:
Merge pull request #427 from acsone/10.0-bank-account-delete-ape

[IMP] account_payment_partner: Prevent to delete bank account linked on invoices.

etobella committed:
[IMP] account_payment_order: report show the invoice number or the move name

OCA-git-bot committed:
[UPD] addons table in README.md [ci skip]

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update account_payment_purchase.pot

pedrobaeza committed:
[FIX] account_payment_order: Fix FR translation

oca-travis committed:
[UPD] Update account_banking_mandate.pot

oca-travis committed:
[UPD] Update account_banking_mandate_sale.pot

oca-travis committed:
[UPD] Update account_banking_pain_base.pot

oca-travis committed:
[UPD] Update account_banking_sepa_credit_transfer.pot

oca-travis committed:
[UPD] Update account_banking_sepa_direct_debit.pot

oca-travis committed:
[UPD] Update account_payment_mode.pot

oca-travis committed:
[UPD] Update account_payment_order.pot

oca-travis committed:
[UPD] Update account_payment_partner.pot

oca-travis committed:
[UPD] Update account_payment_sale.pot

hhgabelgaard committed:
Translated using Weblate (Danish (da_DK))

Currently translated at 28.1% (54 of 192 strings)

Translation: bank-payment-10.0/bank-payment-10.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-10-0/bank-payment-10-0-account_payment_order/da_DK/

hhgabelgaard committed:
Translated using Weblate (Danish (da_DK))

Currently translated at 30.8% (4 of 13 strings)

Translation: bank-payment-10.0/bank-payment-10.0-account_payment_partner
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-10-0/bank-payment-10-0-account_payment_partner/da_DK/

ovnicraft committed:
[10.0][MIG] payment killer migration (#480)

oca-travis committed:
[UPD] Update account_voucher_killer.pot

OCA-git-bot committed:
[UPD] addons table in README.md [ci skip]

OCA-git-bot committed:
[ADD] setup.py

pedrobaeza committed:
Merge pull request #433 from CompassionCH/10.0-payment-cancel

[10.0] [ADD] payment_line cancel module

oca-travis committed:
[UPD] Update account_payment_line_cancel.pot

oca-travis committed:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
[UPD] addons table in README.md [ci skip]

OCA-git-bot committed:
[ADD] setup.py

pedrobaeza commented:
It seems CdtrSchmeId is only for France and England, isn't it?

It's better to have each field addition on one commit for better isolation.

You have to check PEP8 errors:
```
account_payment_order/models/bank_payment_line.py:62:13: E126 continuation line over-indented for hanging indent
account_payment_order/models/bank_payment_line.py:71:9: E121 continuation line under-indented for hanging indent
```

andreparames commented:
> It seems CdtrSchmeId is only for France and England, isn't it?

I  believe those are just the languages in which that page is translated; the field is, as far as I can tell, part of the standard SEPA format, and at least some banks and companies from outside those countries use it.

> It's better to have each field addition on one commit for better isolation. You have to check PEP8 errors

Thanks, I'll restructure and push again.

andreparames committed:
[ADD] sepa: support Initiating Party Scheme

sbidoul committed:
[ADD] sepa: support for Purpose Code

Abranes committed:
[MIG] account_payment_purchase: Migration to 10.0

oca-travis committed:
[UPD] Update account_payment_purchase.pot

thomaspot committed:
Translated using Weblate (Nederlands (nl_NL))

Currently translated at 100,0% (9 of 9 strings)

Translation: bank-payment-10.0/bank-payment-10.0-account_payment_purchase
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-10-0/bank-payment-10-0-account_payment_purchase/nl_NL/

OCA-git-bot committed:
[UPD] addons table in README.md [ci skip]

OCA-git-bot committed:
[ADD] setup.py

andreparames committed:
[ADD] sepa: add Instruction Identification field

Same as EndToEndId

sbidoul commented:
@andreparames could you look into the travis errors: it seems there is a schema validation error.

sbidoul commented on account_banking_pain_base/models/account_payment_mode.py:
```diff
@@ -27,3 +27,8 @@ class AccountPaymentMode(models.Model):
+        string='Initiating Party Schema', size=35,
+        help="This will be used as the 'Initiating Party Schema Name' in "
+        "the PAIN files generated by Odoo. If not defined, no schema will "
+        "be used.\n")
```
It would be nice to have some more help (such as example values, and possibly a pointer to the specification).

sbidoul commented on account_payment_order/models/bank_payment_line.py:
```diff
@@ -58,6 +58,17 @@ class BankPaymentLine(models.Model):
+        ("CASH", "Cash Management transfer"),
+        ("SUPP", "Other payment"),
+        ("NETT", "Netting"),
+    ], string="Purpose Code")
```
Don't you need to add this purpose code on `account.payment.line`, and use it as [grouping criteria](https://github.com/OCA/bank-payment/blob/9d0c6e1c3d673b8eb8040162158fef7dd561083b/account_payment_order/models/bank_payment_line.py#L62) to generate the `bank.payment.line`? Indeed, users don't normally generate bank payment lines manually.

StefanRijnhart committed:
[ADD] Allow for multiple SEPA payment methods with different versions (#493) (#496)

Commit:
Read access on account.payment.method to employees (instead of Invoicing group) (#488)

oca-travis committed:
[UPD] Update account_banking_pain_base.pot

andreparames commented:
I've improved the help description, but I couldn't find a good link to point people to.

sbidoul reviewed (approved):
Looks good (code review)

pedrobaeza commented on account_payment_order/models/bank_payment_line.py:
```diff
@@ -58,6 +58,17 @@ class BankPaymentLine(models.Model):
         'res.currency',
         related='order_id.payment_mode_id.company_id.currency_id',
         readonly=True, store=True)
+    purpose_code = fields.Selection(selection=[
```
Instead of repeating the same selection, use the same trick as we do in sale_order_type:

https://github.com/OCA/sale-workflow/blob/e1149e3f39de09cf1ce5094f9ba1845d3e43eadd/sale_order_type/models/sale_order_type.py#L15-L18

Commit:
[10.0] Add filter on partners on wizard "Create Payment Lines from Journal Items" (#489)

* Add filter on partners on wizard "Create Payment Lines from Journal Items"

* Bump version number

oca-travis committed:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
[UPD] addons table in README.md [ci skip]

ecino committed:
[IMP] account_payment_order: Use bank.payment.line reference in Payment Order Move Lines (#418)

oca-travis committed:
[UPD] Update account_payment_order.pot

sbidoul committed:
[IMP] pain: add hook for party identifiers
